core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -3671,7 +3671,7 @@ class DAGSchedulerSuite extends SparkFunSuite with TempLocalSparkContext with Ti
 
     completeShuffleMapStageSuccessfully(0, 0, parts)
     val shuffleStage = scheduler.stageIdToStage(0).asInstanceOf[ShuffleMapStage]
-    assert(!shuffleStage.shuffleDep.shuffleMergeEnabled)
+    assert(shuffleStage.shuffleDep.mergerLocs.isEmpty)
```

Comment:
check for `!shuffleMergeAllowed` in addition to empty mergers here ?

Reply:
Looks like this was missed ?

Reply:
Addressed it now. Sorry probably in the refactor, it went back.

Reply:
I think here the `shuffleMergeAllowed` won't be false. Here the mergers are not set due to not enough mergers, so it doesn't take push based shuffle path.

Reply:
You are right, my bad.